### PR TITLE
Capacity change algorithm rewrite, closes #192

### LINKED ISSUE #192: BUG: Capacity change algorithm provides incorrect answer under certain circumstances
**Describe the bug**
When using the default (non-MOSEK) solver in the main pipeline, the capacity change algorithm returns incorrect results on some test data with known capacity changes. Specifically, the trend term is large and positive, fitting the capacity changes with the trend rather than the piecewise constant term. This only occurs for some edge cases of true positive data sets.

Image

**To Reproduce**
1. Load PVO test data provided by [Chris Deline (NREL)](https://github.com/cdeline)
2. Run data handler pipeline with default solver
3. Inspect capacity change results

**Expected behavior**
When solving with MOSEK, you get the following result, with is expected:

Image

**Environment (please complete the following information about your working environment):**
 - OS: MacOS
 - Python version: 3.12.3
 - Solar Data Tools version: 1.6.6
 - cvxpy: 1.6.2
 - clarabel: 0.9.0
 - mosek: 11.0.8

### PR DESCRIPTION
**Pull request by bmeyers**
Closes #192 

This PR addresses the bug that was uncovered where the capacity change algorithm returns incorrect results on some test data when using the default solver. This is a rewrite of the signal decomposition implementation, such that the exact some problem is now being solved by clarabel and mosek. 

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] Updated or added relevant tests
- [x] Updated relevant documentation
- [x] Added relevant label(s)
- [x] All comments are resolved

### CODE REVIEW

**Comment by bmeyers** [comment]
Just added small change to the polar transform plotting function to fix #188

File: solardatatools/signal_decompositions.py [hunk 1]
```diff
@@ -193,57 +195,89 @@ def tl1_l2d2p365(
...
-        # MOSEK weights set in CVXPY function
-        res = _cvx_l1_l1d1_l2d2p365(
-            signal=signal,
-            use_ixs=use_ixs,
-            return_all=return_all,
-            solver=solver,
-            verbose=verbose,
-        )
+    if solver is not None:
+        if solver.upper() not in ["MOSEK", "CLARABEL", "OSQP"]:
```
[the remark this answers is not in the document]
Comment (bmeyers) [reply solardatatools/signal_decompositions.py:224]: We should remove OSQP as well

**Comment by bmeyers** [comment]
@pluflou all checks are passing again, so I think we're ready for the final review

**Review by pluflou — APPROVED** [review]
